AstarteDeviceSDKCSharp/Utilities/DecodedMessage.cs

```diff
@@ -54,6 +54,50 @@ public void SetTimestamp(DateTime timestamp)
         {
             this.timestamp = timestamp;
         }
+
+        public bool PayloadEquality(object? otherPayload)
+        {
+            if (otherPayload is null)
+            {
+                return false;
+            }
+
+            if (payload is Array array && otherPayload is Array otherArray)
+            {
+                if (array == null || otherArray == null) return false;
+                if (array.Length != otherArray.Length) return false;
+                return ArrayEquality(array, otherArray);
```

Comment:
What do you think about using [Enumerable.SequenceEqual](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?redirectedfrom=MSDN&view=net-7.0#System_Linq_Enumerable_SequenceEqual__1_System_Collections_Generic_IEnumerable___0__System_Collections_Generic_IEnumerable___0__) ? 
```suggestion
                return Enumerable.SequenceEqual(array, otherArray);
```

Reply:
We don't know TSource at this point. We must cast both arrays to their data type to implement this method. 
(etc. `return Enumerable.SequenceEqual((IEnumerable<int>)array, (IEnumerable<int>)otherArray)`)